ui/app/components/recovery/page/snapshots/index.hbs

```diff
@@ -3,4 +3,58 @@
   SPDX-License-Identifier: BUSL-1.1
 }}
 
-{{! TODO: This is an empty skeleton component which will be completed as part of 1.21 feature work on single item recovery }}
\ No newline at end of file
+<Hds::PageHeader class="has-top-padding-l has-bottom-padding-m" as |PH|>
+  <PH.Title>
+    Secrets Recovery
+    {{#if this.version.isCommunity}}
+      <Hds::Badge @text="Enterprise" @color="highlight" data-test-badge="enterprise" />
+    {{/if}}
+  </PH.Title>
+
+  <PH.Subtitle>
+    Recover lost or deleted secrets from a raft snapshot. Supported types include KV v1 and Cubbyhole.
+  </PH.Subtitle>
+</Hds::PageHeader>
+
+{{#if (eq this.loadedSnapshots.length 0)}}
+  {{#let (get this.emptyStateDetails this.state) as |d|}}
+    <EmptyState @title={{d.title}} @icon={{d.icon}} @message={{d.message}}>
+
+      {{#if (eq this.state this.viewState.ALLOW_UPLOAD)}}
+        <Hds::Button @text={{d.buttonText}} @color={{d.buttonColor}} {{on "click" this.uploadSnapshot}} />
+      {{else}}
+        <Hds::Button
+          @text={{d.buttonText}}
+          @color={{d.buttonColor}}
+          @icon={{d.buttonIcon}}
+          @iconPosition="trailing"
+          @route={{d.buttonRoute}}
+          @query={{if d.buttonRoute (hash namespace="")}}
+          @href={{doc-link d.buttonHref}}
+        />
+      {{/if}}
+    </EmptyState>
+  {{/let}}
+{{else}}
+
+  <div class="loaded-snapshot">
+    {{#each this.loadedSnapshots as |s index|}}
+      {{#if (eq index 0)}}
+        <Hds::Card::Container class="snapshot-card-container" @level="mid" @hasBorder={{true}}>
+          <div class="flex-row">
+            <Hds::Text::Display @tag="h2">Snapshot</Hds::Text::Display>
+            <Hds::Badge @text={{s.status}} @color="success" data-test-badge="status" />
+          </div>
+          <Hds::Text::Body @tag="p">Expiring at {{s.expires_at}}</Hds::Text::Body>
+          <Hds::Link::Standalone
+            @iconPosition="trailing"
+            @icon="arrow-right"
+            @text="View details"
+            @route="vault.cluster.recovery.snapshots.snapshot"
+            @model={{s.snapshot_id}}
+          />
+        </Hds::Card::Container>
```

Comment:
This will actually live in the snapshots.snapshot route since we're looking at a single snapshot there